src/catalog/metastore.rs

```diff
@@ -70,9 +73,18 @@ impl Metastore {
             functions: external_store,
             staging_schema,
             default_store,
+            object_store_cache: None,
         }
     }
 
+    pub fn with_object_store_cache(
+        mut self,
```

Comment:
(a bit more idiomatic)

```suggestion
        &mut self,
```

Reply:
I guess it's down to preference whether the builder consumes/returns a thing, or updates the reference of the thing: https://users.rust-lang.org/t/builder-pattern-in-rust-self-vs-mut-self-and-method-vs-associated-function/72892/4

I was mostly guided by the object_store crate which goes with the former approach.